ticketbuyer/buyer.go

```diff
@@ -308,14 +486,11 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
 	// Set the max price to the configuration parameter that is lower
 	// Absolute or relative max price
 	var maxPriceAmt dcrutil.Amount
-	if t.cfg.MaxPriceAbsolute > 0 && t.cfg.MaxPriceAbsolute < avgPrice*t.cfg.MaxPriceRelative {
-		maxPriceAmt, err = dcrutil.NewAmount(t.cfg.MaxPriceAbsolute)
-		if err != nil {
-			return ps, err
-		}
+	if t.MaxPriceAbsolute().ToCoin() > 0 && t.MaxPriceAbsolute().ToCoin() < avgPrice*t.MaxPriceRelative() {
+		maxPriceAmt = t.MaxPriceAbsolute()
```

Comment:
this is incorrect, read the value once and do all checking on it before using it.